src/benchmark/operators/hash_join_benchmark.cpp

```diff
@@ -0,0 +1,757 @@
+      std::vector<JoinPredicate>(additional_predicates));
+  latest_operator->execute();
+
+  // Print::print(latest_operator);
+
+  /*std::cout << "Row count after predicates: "
+            << latest_operator->get_output()->row_count() << std::endl;*/
+}
+
+/**
+ * Creates a benchmark set for a multi predicate join. table 1 will contain fact_table_size many rows.
+ * Table 2 will contain fact_factor * probing_factor * fact_table_size many rows,
+ * The intermediate result (after the first join) will contain fact_factor * fact_factor * probing_factor * fact_table_size many results.
+ * The joined table will contain fact_factor * probing_factor rows again.
+ */
+void execute_multi_predicate_join(benchmark::State& state, size_t chunk_size, size_t fact_table_size,
+                                  size_t fact_factor, double probing_factor, bool with_scan,
+                                  bool as_reference_segment) {
+  ColumnGenerator gen;
+
+  const auto join_pair =
+      gen.generate_two_predicate_join_tables(chunk_size, fact_table_size, fact_factor, probing_factor);
+
+  // std::cout << join_pair->first->row_count() << std::endl;
+  // std::cout << join_pair->second->row_count() << std::endl;
+
+  const auto table_wrapper_left = std::make_shared<TableWrapper>(join_pair->first);
+  table_wrapper_left->execute();
+  const auto table_wrapper_right = std::make_shared<TableWrapper>(join_pair->second);
+  table_wrapper_right->execute();
+
+  std::vector<JoinPredicate> join_predicates;
+  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+  //  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{1}, ColumnID{1}}, PredicateCondition::Equals});
+
+  clear_cache();
+
+  //  Print::print(table_wrapper_left);
+  //  Print::print(table_wrapper_right);
+
+  std::shared_ptr<AbstractOperator> validate_left;
+  std::shared_ptr<AbstractOperator> validate_right;
+
+  if (as_reference_segment) {
+    // validate needs a transaction context. This is taken from test/operators/validate_test.cpp:60
+    auto context = std::make_shared<TransactionContext>(1u, 3u);
+
+    // src/lib/operators/validate.cpp:121
+    validate_left = std::make_shared<Validate>(table_wrapper_left);
+    validate_right = std::make_shared<Validate>(table_wrapper_right);
+
+    validate_left->set_transaction_context(context);
+    validate_right->set_transaction_context(context);
+
+    validate_left->execute();
+    validate_right->execute();
+  } else {
+    validate_left = table_wrapper_left;
+    validate_right = table_wrapper_right;
+  }
+
+  // Warm up
+  if (with_scan) {
+    execute_multi_predicate_join_with_scan(validate_left, validate_right, JoinMode::Inner, join_predicates);
+  } else {
+    execute_multi_predicate_join(validate_left, validate_right, JoinMode::Inner, join_predicates);
+  }
+
+  while (state.KeepRunning()) {
```

Comment:
Please use the for-based loop of C++11 which Google recommends to use for gbench.